### ISSUE TITLE
PHP Recoverable fatal error:  Object of class DateTime could not be converted to string in ..../vendor/guzzlehttp/psr7/src/Query.php on line 96

### ISSUE THREAD

**Issue by Vitexus** [FitecoGroup/php-kimai-api-client#5]
Hallo the documentation promise use of \DateTime objects: https://github.com/FitecoGroup/php-kimai-api-client/blob/master/docs/Api/TimesheetApi.md#apiTimesheetsGet

But there is no conversion to string for Guzzle. 

Please update documentation or add some ->format( to proper places. 

Thank you for your time. 

```
"/usr/bin/php" "/home/vitex/Projects/Sizek/kimai2abraflexi/src/kiami2abraflexi.php"
PHP Recoverable fatal error:  Object of class DateTime could not be converted to string in /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/guzzlehttp/psr7/src/Query.php on line 96
PHP Stack trace:
PHP   1. {main}() /home/vitex/Projects/Sizek/kimai2abraflexi/src/kiami2abraflexi.php:0

PHP   2. Kimai2AbraFlexi\Importer->__construct($init = *uninitialized*, $options = *uninitialized*) /home/vitex/Projects/Sizek/kimai2abraflexi/src/kiami2abraflexi.php:22
PHP   3. Fiteco\KimaiClient\Api\TimesheetApi->apiTimesheetsGet($user = NULL, $customer = NULL, $customers = NULL, $project = NULL, $projects = NULL, $activity = NULL, $activities = NULL, $page = 1, $size = 100, $tags = NULL, $orderBy = NULL, $order = NULL, $begin = class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $end = class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $exported = *uninitialized*, $active = *uninitialized*, $full = *uninitialized*, $term = *uninitialized*, $modifiedAfter = *uninitialized*) /home/vitex/Projects/Sizek/kimai2abraflexi/src/Kimai2AbraFlexi/Importer.php:82
PHP   4. Fiteco\KimaiClient\Api\TimesheetApi->apiTimesheetsGetWithHttpInfo($user = NULL, $customer = NULL, $customers = NULL, $project = NULL, $projects = NULL, $activity = NULL, $activities = NULL, $page = 1, $size = 100, $tags = NULL, $orderBy = NULL, $order = NULL, $begin = class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $end = class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $exported = NULL, $active = NULL, $full = NULL, $term = NULL, $modifiedAfter = NULL) /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/fiteco-group/kimai-client/src/Api/TimesheetApi.php:380
PHP   5. Fiteco\KimaiClient\Api\TimesheetApi->apiTimesheetsGetRequest($user = NULL, $customer = NULL, $customers = NULL, $project = NULL, $projects = NULL, $activity = NULL, $activities = NULL, $page = 1, $size = 100, $tags = NULL, $orderBy = NULL, $order = NULL, $begin = class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $end = class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $exported = NULL, $active = NULL, $full = NULL, $term = NULL, $modifiedAfter = NULL) /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/fiteco-group/kimai-client/src/Api/TimesheetApi.php:417
Recoverable fatal error: Object of class DateTime could not be converted to string in /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/guzzlehttp/psr7/src/Query.php on line 96

Call Stack:
PHP   6. GuzzleHttp\Psr7\Query::build($params = ['page' => 1, 'size' => 100, 'begin' => class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, 'end' => class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }], $encoding = *uninitialized*) /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/fiteco-group/kimai-client/src/Api/TimesheetApi.php:858
    0.1058    1508000   1. {main}() /home/vitex/Projects/Sizek/kimai2abraflexi/src/kiami2abraflexi.php:0
    1.9170    2864256   2. Kimai2AbraFlexi\Importer->__construct($init = ???, $options = ???) /home/vitex/Projects/Sizek/kimai2abraflexi/src/kiami2abraflexi.php:22
    8.0684    4060520   3. Fiteco\KimaiClient\Api\TimesheetApi->apiTimesheetsGet($user = NULL, $customer = NULL, $customers = NULL, $project = NULL, $projects = NULL, $activity = NULL, $activities = NULL, $page = 1, $size = 100, $tags = NULL, $orderBy = NULL, $order = NULL, $begin = class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $end = class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $exported = ???, $active = ???, $full = ???, $term = ???, $modifiedAfter = ???) /home/vitex/Projects/Sizek/kimai2abraflexi/src/Kimai2AbraFlexi/Importer.php:82
    9.4752    4060896   4. Fiteco\KimaiClient\Api\TimesheetApi->apiTimesheetsGetWithHttpInfo($user = NULL, $customer = NULL, $customers = NULL, $project = NULL, $projects = NULL, $activity = NULL, $activities = NULL, $page = 1, $size = 100, $tags = NULL, $orderBy = NULL, $order = NULL, $begin = class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $end = class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $exported = NULL, $active = NULL, $full = NULL, $term = NULL, $modifiedAfter = NULL) /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/fiteco-group/kimai-client/src/Api/TimesheetApi.php:380
    9.4752    4060896   5. Fiteco\KimaiClient\Api\TimesheetApi->apiTimesheetsGetRequest($user = NULL, $customer = NULL, $customers = NULL, $project = NULL, $projects = NULL, $activity = NULL, $activities = NULL, $page = 1, $size = 100, $tags = NULL, $orderBy = NULL, $order = NULL, $begin = class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $end = class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, $exported = NULL, $active = NULL, $full = NULL, $term = NULL, $modifiedAfter = NULL) /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/fiteco-group/kimai-client/src/Api/TimesheetApi.php:417
    9.4759    4080216   6. GuzzleHttp\Psr7\Query::build($params = ['page' => 1, 'size' => 100, 'begin' => class DateTime { public $date = '2021-07-01 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }, 'end' => class DateTime { public $date = '2021-07-31 00:00:00.000000'; public $timezone_type = 3; public $timezone = 'Europe/Berlin' }], $encoding = ???) /home/vitex/Projects/Sizek/kimai2abraflexi/vendor/fiteco-group/kimai-client/src/Api/TimesheetApi.php:858

Done.
```

My composer.json:

```json
{
    "name": "vitexsoftware/kimai2abraflexi",
    "description": "Generate invoice from Kimai timesheet",
    "type": "php-abraflexi",
    "license": "MIT",
    "authors": [
        {
            "name": "Vítězslav Dvořák",
            "email": "info@vitexsoftware.cz"
        }
    ],
    "minimum-stability": "dev",
    "require": {
        "fiteco-group/kimai-client": "0.2.0",
        "spojenet/flexibee": "2.0.x-dev",
        "guzzlehttp/psr7": "2.0.x-dev"
    },
    "autoload": {
        "psr-4": {
            "Kimai2AbraFlexi\\": "src/Kimai2AbraFlexi/",
            "Test\\Kimai2AbraFlexi\\": "testing/src/Kimai2AbraFlexi/"
        }
    }
}

```

**Comment by NikoGrano** [comment]
Well, this is good question how to handle non-string objects which do not implement `__toString()` function. This should be anyways be fixed in level of `openapi-generator`.

I think you can meanwhile just use `->format(DATETIME_ATOM)` or is it working? I could update docs to that.
Or create wrapper object which would implement `__toString`

**Comment by Vitexus** [comment]
The best solution (for me) is to be ready Accept \DateTime object AND timestamp string by https://www.kimai.org/documentation/rest-api.html#datetime-formats-iso-8601-and-html5.

**Comment by NikoGrano** [comment]
Fixed in 0.2.1
